receiver/datadogreceiver/factory.go

```diff
@@ -46,11 +48,13 @@ func createDefaultConfig() component.Config {
 	}
 }
 
-func createTracesReceiver(
-	ctx context.Context,
-	params receiver.CreateSettings,
-	cfg component.Config,
-	consumer consumer.Traces) (receiver.Traces, error) {
+func createTracesReceiver(ctx context.Context, params receiver.CreateSettings, cfg component.Config, consumer consumer.Traces) (r receiver.Traces, err error) {
 	rcfg := cfg.(*Config)
-	return newDataDogReceiver(rcfg, consumer, params)
+	r = receivers.GetOrAdd(cfg, func() component.Component {
+		dd, _ := newDataDogReceiver(rcfg, consumer, params)
+		return dd
+	})
+	return r, nil
 }
+
+var receivers = sharedcomponent.NewSharedComponents()
```

Comment:
Is this pattern being used elsewhere? Is this new?

Reply:
No its been around and a lot of receivers and exports use it